### DISCUSSION TITLE
Deploying seq container got "please retry later" from index.docker.io

### DISCUSSION THREAD

**Discussion by rodneytamblyn** [datalust/seq-tickets#2223] [General]
Attempting to deploy Seq container today in Azure I got this docker registry error.  Anyone run across this before or can shed light?  I will assume it means what it says and try again later...

```
--image datalust/seq:latest   ....

(RegistryErrorResponse) An error response is received from the docker registry 'index.docker.io'. Please retry later.
Code: RegistryErrorResponse
Message: An error response is received from the docker registry 'index.docker.io'. Please retry later.
```

**Comment by rodneytamblyn** [comment]
And deploying via yaml
```
rodney@moonlight:~/seq-logging$ az container create --verbose --resource-group ap-seq-aci --file config-ap.yaml 
(MultipleErrorsOccurred) Multiple error occurred: 'Conflict':'RegistryErrorResponse':'An error response is received from the docker registry 'index.docker.io'. Please retry later.';'Conflict':'RegistryErrorResponse':'An error response is received from the docker registry 'index.docker.io'. Please retry later.';'Conflict':'RegistryErrorResponse':'An error response is received from the docker registry 'index.docker.io'. Please retry later.'.
Code: MultipleErrorsOccurred
```

**Comment by rodneytamblyn** [comment]
A few hours later.... now seems to be working.  No changes to configs, so maybe the error "try again later" was actually correct.  I see on Docker that the image was updated today, maybe that was the cause....

https://hub.docker.com/r/datalust/seq
![image](https://github.com/datalust/seq-tickets/assets/1562956/20ef245c-60af-4c17-b6f5-137d40af0ea0)

**Reply by george-ognyanov-kolev** [reply]
Facing the same issue and seeing how your post is the only result when I look for the error is discouraging. I am pulling from a personal repo so the image being updated is not applicable. Have you tried deploying with another docker image besides this? If I deploy with a "starter image" from Azure Container Instance there is no issue on my end.

**Reply by rodneytamblyn** [reply]
HI George,
I thought about pulling the image and pushing to my own container repository to see if it would work better.  Given you're experiencing the issue too, I suspect that the update to the container images that (appears) to have been pushed by datalust yesterday must have broken something...

Have you tried deploying container using one of the other tags (other than 'latest')?

Rod

**Reply by abarqawi** [reply]
https://medium.com/@alaa.barqawi/docker-rate-limit-with-azure-container-instance-and-aks-4449cede66dd

**Reply by george-ognyanov-kolev** [reply]
@abarqawi nice one. This kinda explains whats going on. 

@rodneytamblyn I have tried using another tag but sadly it didn't work either. Eventually, I just decided to delete the entire repo and create it again with either latest / newest tags but still didn't do anything. I even tried pulling the nginx image instead of the one from my repo but with no avail. Really wonder what the issue is since I am nowhere close to the 6-hour quotas mentioned by @abarqawi.

**Reply by george-ognyanov-kolev** [reply]
switching from public to private docker repo seemed to fix the problem for me

**Comment by abarqawi** [comment]
@rodneytamblyn 
this related to rate limiting for Azure & Docker hub 
https://medium.com/@alaa.barqawi/docker-rate-limit-with-azure-container-instance-and-aks-4449cede66dd

**Reply by rodneytamblyn** [reply]
Yep, this definitely looks like the explanation.  Docker are not really helping with their error message which doesn't include the keyword "rate limited" which might provide some clues, let alone advice on how to fix.

**Comment by rodneytamblyn** [comment]
So the best approach would be to pull the container and push to your own container repository and use that (if you don't have a dockerhub paid account).  This might need to become standard operating procedure.

e.g. https://stackoverflow.com/questions/28349392/how-to-push-a-docker-image-to-a-private-repository

**Reply by abarqawi** [reply]
Yes

**Comment by rodneytamblyn** [comment]
Agree re quotas / limits.  I have raised a ticket with Azure.  I suggest pull the container and tag/save it to your own private container repo, then use that.  If you encounter this error keep retrying periodically, it eventually worked for me.

R.

**Comment by KodrAus** [comment]
This looks unrelated to Seq itself so will close the discussion here, but it's definitely interesting to keep in mind :+1:

**Comment by rodneytamblyn** [comment]
Ashley,

The container deployment instructions page should be updated to warn people about this issue.  Even if they do successfully initially deploy their instance it could fail at any time - for example if Azure redeploys their container and the pull fails.  Not what you would expect but this is what this rate limiting will result in. 

Warm Regards,

Rodney
